src/gui/shareusergroupwidget.cpp

```diff
@@ -926,13 +932,46 @@ void ShareUserLine::customizeStyle()
     _deleteShareButton->setIcon(deleteicon);
 
     _ui->noteConfirmButton->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
-    _ui->confirmExpirationDate->setIcon(Theme::createColorAwareIcon(":/client/theme/confirm.svg"));
     _ui->progressIndicator->setColor(QGuiApplication::palette().color(QPalette::WindowText));
 
     // make sure to force BackgroundRole to QPalette::WindowText for a lable, because it's parent always has a different role set that applies to children unless customized
     _ui->errorLabel->setBackgroundRole(QPalette::WindowText);
 }
 
+QPixmap ShareUserLine::pixmapForShareeType(int type, const QColor &backgroundColor) const
+{
+    const QString pixmapColor = backgroundColor.isValid() && !Theme::isDarkColor(backgroundColor) ? "black" : "white";
+    switch (type) {
+    case Sharee::Room:
+        return QPixmap::fromImage(QImage(QString(":/client/theme/%1/talk-app.svg").arg(pixmapColor)));
+    case Sharee::Email:
+        return QPixmap::fromImage(QImage(QString(":/client/theme/%1/email.svg").arg(pixmapColor)));
```

Comment:
Why not create a svg in the matching color on the fly? We could reuse that function for example in the main dialog if we ever have dark mode support.
You can use QSvgRenderer for that. If you need some inspiration have a look at my closed pr https://github.com/nextcloud/desktop/pull/3026/files in the function `ColorSvgImageProvider::requestImage()`.

I would suggest we create a function in theme.cpp that takes a svg picture name and a color as argument and then creates the svg with the given color.

Reply:
We could also make the function that renders the svg picture into a extra class and add to that class some cache so that the picture does not need to be rendered all the time again.